test/test-jsdomparser.js

```diff
@@ -6,7 +6,7 @@ var readability = require("../index.js");
 var JSDOMParser = readability.JSDOMParser;
 
 var BASETESTCASE = '<html><body><p>Some text and <a class="someclass" href="#">a link</a></p>' +
-                   '<div id="foo">With a <script>With < fancy " characters in it because' +
+                   '<div id="foo">With a <script>With &lt; fancy " characters in it because' +
```

Comment:
I replace all `<` and `>` with `&lt;` and `&gt;` because of the below three reasons.

1. Our [xml serializer][xml-serializer] just replaces all `<` and `>` with `&lt;` and `&gt;`. And the output of the xml serializer is just the input of reader mode.
2. In XHTML, the script and style elements are declared as having #CDATA content. [1]
3. Use external scripts if your script uses < or & or ]]> or --. [2] So replace `<` and `>` in embedded script.

Due to above reasons, I think we might need to change these `source.html` pages. What do you think?

[1]: https://www.w3.org/TR/xhtml1/#h-4.8
[2]: https://www.w3.org/TR/xhtml1/#C_4
[xml-serializer]: http://searchfox.org/mozilla-central/source/toolkit/components/reader/ReaderMode.jsm#442